katdal/chunkstore_s3.py

```diff
@@ -446,6 +465,15 @@ def create_array(self, array_name):
             with self.request(None, 'PUT', policy_url, data=json.dumps(policy)):
                 pass
 
+        if self.expiry_days > 0:
+            lifecycle_url = urllib.parse.urljoin(url, '?lifecycle')
+            lifecycle = copy.deepcopy(_BASE_LIFECYCLE_POLICY)
+            xml_payload = lifecycle.format(self.expiry_days)
+            b64_md5 = base64.encodestring(hashlib.md5(xml_payload.encode('utf-8')).digest()).decode('utf-8')[:-1]
```

Comment:
Any reason to be using `encodestring` rather than `b64encode`? From what I can see, the difference is that it's going to insert newlines, but I'm assuming `[:-1]` is to undo that.